src/stores/widgets/StopGapWidgetDriver.ts

```diff
@@ -271,20 +276,20 @@ export class StopGapWidgetDriver extends WidgetDriver {
     public async sendEvent<K extends keyof StateEvents>(
         eventType: K,
         content: StateEvents[K],
-        stateKey?: string,
-        targetRoomId?: string,
+        stateKey: string | null,
+        targetRoomId: string | null,
     ): Promise<ISendEventDetails>;
     public async sendEvent<K extends keyof TimelineEvents>(
         eventType: K,
         content: TimelineEvents[K],
         stateKey: null,
-        targetRoomId?: string,
+        targetRoomId: string | null,
     ): Promise<ISendEventDetails>;
     public async sendEvent(
         eventType: string,
         content: IContent,
-        stateKey?: string | null,
-        targetRoomId?: string,
+        stateKey: string | null = null,
+        targetRoomId: string | null = null,
```

Comment:
why the changes from undefined to null? I think in general intend want to move in the other direction.

Reply:
It's to match the signature of `WidgetApi#sendEvent`, which uses `string | null = null` for these arguments.

It also struck me as dangerous to allow `stateKey` to be `undefined` or `null` when there was only a `!== null` check on it, and felt that restricting both parameters to just `null` made things simpler.

Reply:
ok, it would probably be good to change the widget api at some point but that needn't be today